File: v1-nestjs/server/src/search/search.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { SearchService } from './search.service';

@Controller('search')
export class SearchController {
  constructor(private readonly searchService: SearchService) {}

  @Get()
  async searchAll(@Query('q') q: string) {
    const results = await this.searchService.searchAll(q || '');
    return { code: 0, data: results };
  }
}

File: syncflow-java/syncflow-common/src/main/java/com/syncflow/common/controller/resource/SearchController.java
package com.syncflow.common.controller.resource;

import com.syncflow.common.dto.search.SearchResultVO;
import com.syncflow.common.result.Result;
import com.syncflow.common.service.resource.SearchService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * Global search controller.
 */
@RestController
@RequestMapping("/api/search")
public class SearchController {

    private final SearchService searchService;

    public SearchController(SearchService searchService) {
        this.searchService = searchService;
    }

    /**
     * Global search across articles, templates, resources.
     */
    @GetMapping
    public Result<SearchResultVO> search(@RequestParam String q) {
        SearchResultVO result = searchService.search(q);
        return Result.success(result);
    }
}
